### PR TITLE
[CALCITE-5178] Single column with ROW type generates wrong plan

### PR DESCRIPTION
**Pull request by libenchao**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/adapter/enumerable/JavaRowFormat.java [hunk 1]
```diff
@@ -82,6 +83,11 @@ public enum JavaRowFormat {
         JavaTypeFactory typeFactory,
         RelDataType type) {
       assert type.getFieldCount() == 1;
+      RelDataType field0Type = type.getFieldList().get(0).getType();
```
Comment (JiajunBernoulli) [inline core/src/main/java/org/apache/calcite/adapter/enumerable/JavaRowFormat.java:86]: Does the code can be simplified?
```      
      RelDataType field0Type = type.getFieldList().get(0).getType();
      // nested ROW type is always represented as array.
      return field0Type.getSqlTypeName() == SqlTypeName.ROW
          ? Object[].class
          : typeFactory.getJavaClass(field0Type);
    ```

Comment (libenchao) [reply core/src/main/java/org/apache/calcite/adapter/enumerable/JavaRowFormat.java:86]: @JiajunBernoulli Thanks for your review. I don't have a preference for either way actually.

**Review by chunweilei — APPROVED** [review]
LGTM

**Comment by libenchao** [comment]
merging now.

**Pull request closed without merging**